cmd/lakectl/cmd/store/s3.go

```diff
@@ -12,10 +12,23 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
 )
 
-func GetS3Client() (*s3.S3, error) {
-	sess, err := session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	})
+func GetS3Client(s3EndpointUrl string) (*s3.S3, error) {
+	var sess *session.Session
+	var err error
+	if len(s3EndpointUrl) != 0 {
+		sess, err = session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{
+				Endpoint:         aws.String(s3EndpointUrl),
+				Region:           aws.String("us-east-1"), // Needs region for validation as it is AWS client
+				S3ForcePathStyle: aws.Bool(true),
+			},
+			SharedConfigState: session.SharedConfigEnable,
+		})
+	} else {
+		sess, err = session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		})
+	}
```

Comment:
(Not required!)

Usually I prefer to extract shared code outside of conditionals.  Here this would be dropping the :
```suggestion
	var config aws.Config{}
	if s3EndpointUrl != "" {
		config = aws.Config{
			Endpoint:         aws.String(s3EndpointUrl),
			Region:           aws.String("us-east-1"), // Needs region for validation as it is AWS client
			S3ForcePathStyle: aws.Bool(true),
			}
	}
	sess, err := session.NewSessionWithOptions(session.Options{
		SharedConfigState: session.SharedConfigEnable,
		Config: config,
	})
	if err != nil {
		return nil, err
	}
```